File: src/contests/cg/puzzle/swift/rubik.swift
// 🎮 CodinGame Puzzle - rubik
// https://www.codingame.com/training/medium/rubik

let n = Int(readLine()!)!
// Visible mini-cubes = all cubes minus the hidden inner cube of side n-2.
let inner = n >= 2 ? n - 2 : 0
print(n * n * n - inner * inner * inner)

File: src/contests/cg/puzzle/rubik.ts
// 🎮 CodinGame Puzzle - rubik
// https://www.codingame.com/training/medium/rubik

const N: number = parseInt(readline(), 10)
// Visible mini-cubes = all cubes minus the hidden inner cube of side N-2.
const inner = N >= 2 ? N - 2 : 0
console.log(N * N * N - inner * inner * inner)
